From 4de7f03411514867bfcb51cd8752a26f0c354ce9 Mon Sep 17 00:00:00 2001
From: Marcelo de Castro Loebens <Marcelo.DeCastroLoebens@windriver.com>
Date: Fri, 24 Oct 2025 15:03:21 -0400
Subject: [PATCH] Redo patch to reset password when adding user

Avoiding failure with code 20 when including the field "pwdReset",
used to reset the password for the used add.

In trixie, the field seems to be set by default when user is
created. This sugests that this patch could be redundant, but since
I couldn't find the modification that introduced this new behavion,
I'm including the code 20 as a value accepted and keeping the patch.

The original patch was introduced by the change:
I13f098c6053816bb3b0450c039caccf94c04d55d

Signed-off-by: Marcelo de Castro Loebens <Marcelo.DeCastroLoebens@windriver.com>
---
 sbin/ldapadduser | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/sbin/ldapadduser b/sbin/ldapadduser
index cc44f7d..9b9ee63 100755
--- a/sbin/ldapadduser
+++ b/sbin/ldapadduser
@@ -77,6 +77,26 @@ if [ -n "$_PASSWORD" ]
 then
   _changepassword "$_PASSWORD" "uid=$_USER,$USUFFIX,$SUFFIX"
   [ $? -eq 0 ] && echo_log "Successfully set password for user $_USER"
+
+  # reset user's password so the user will be asked to change password.
+  # These variables are used by the runtime script _ldapmodify which is sourced.
+  _ACTION="add"
+  _FIELD="pwdReset"
+  _VALUE="TRUE"
+
+  # Use template if necessary
+  if [ -n "$UMTEMPLATE" ] && [ -r "$UMTEMPLATE" ]
+  then
+    _getldif="cat $UMTEMPLATE"
+  else
+    _getldif="_extractldif 2"
+  fi
+
+  $_getldif | _filterldif | _utf8encode | _ldapmodify
+  ret="$?"
+  # 0 is success, 20 means the value is already set
+  [ $ret -eq 0 ] || [ $ret -eq 20 ] || end_die "Error resetting password for user $_USER"
+  warn_log "Warning : password is reset, user will be asked to change password at login"
 else
   [ -n "$PASSWORDGEN" ] && warn_log "Warning : got invalid password for user $_USER (password not set)"
 fi
-- 
2.34.1

